daemon/src/main/java/bisq/daemon/grpc/GrpcMoneroConnectionsService.java

```diff
@@ -0,0 +1,316 @@
+            return CheckConnectionReply.newBuilder()
+                    .setConnection(toGrpcUriConnection(connection)).build();
+        });
+    }
+
+    @Override
+    public void checkConnections(CheckConnectionsRequest request,
+                                 StreamObserver<CheckConnectionsReply> responseObserver) {
+        handleRequest(responseObserver, () -> {
+            List<UriConnection> connections = coreApi.checkMoneroConnections();
+            List<bisq.proto.grpc.UriConnection> replyConnections = connections.stream()
+                    .map(GrpcMoneroConnectionsService::toGrpcUriConnection).collect(Collectors.toList());
+            return CheckConnectionsReply.newBuilder().addAllConnections(replyConnections).build();
+        });
+    }
+
+    @Override
+    public void startCheckingConnections(StartCheckingConnectionsRequest request,
+                                         StreamObserver<StartCheckingConnectionsReply> responseObserver) {
+        handleRequest(responseObserver, () -> {
+            int refreshMillis = request.getRefreshPeriod();
+            Long refreshPeriod = refreshMillis == 0 ? null : (long) refreshMillis;
+            coreApi.startCheckingMoneroConnection(refreshPeriod);
+            return StartCheckingConnectionsReply.newBuilder().build();
+        });
+    }
+
+    @Override
+    public void stopCheckingConnections(StopCheckingConnectionsRequest request,
+                                        StreamObserver<StopCheckingConnectionsReply> responseObserver) {
+        handleRequest(responseObserver, () -> {
+            coreApi.stopCheckingMoneroConnection();
+            return StopCheckingConnectionsReply.newBuilder().build();
+        });
+    }
+
+    @Override
+    public void getBestAvailableConnection(GetBestAvailableConnectionRequest request,
+                                           StreamObserver<GetBestAvailableConnectionReply> responseObserver) {
+        handleRequest(responseObserver, () -> {
+            UriConnection connection = coreApi.getBestAvailableMoneroConnection();
+            return GetBestAvailableConnectionReply.newBuilder()
+                    .setConnection(toGrpcUriConnection(connection)).build();
+        });
+    }
+
+    @Override
+    public void setAutoSwitch(SetAutoSwitchRequest request,
+                              StreamObserver<SetAutoSwitchReply> responseObserver) {
+        handleRequest(responseObserver, () -> {
+            coreApi.setMoneroConnectionAutoSwitch(request.getAutoSwitch());
+            return SetAutoSwitchReply.newBuilder().build();
+        });
+    }
+
+    private <_Reply> void handleRequest(StreamObserver<_Reply> responseObserver,
+                                        RpcRequestHandler<_Reply> handler) {
+        try {
+            _Reply reply = handler.handleRequest();
+            responseObserver.onNext(reply);
+            responseObserver.onCompleted();
+        } catch (URISyntaxException cause) {
```

Comment:
No need for custom error handling for URI validation. Can use a single catch clause and call `exceptionHandler.handleException(log, cause, responseObserver);` directly.